packages/database/src/entity-repo/User.ts

```diff
@@ -0,0 +1,53 @@
+import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm'
+import { container } from '../data-source'
+import { ModuleRepository } from '../repository/Module'
+import { UserRepository } from '../repository/User'
+import { utils } from '../utils'
+
+@Entity({ name: 'user' })
+export class User {
+  @PrimaryGeneratedColumn('uuid')
+  id: number
+
+  @Column()
+  displayName: string
+
+  @Column()
+  username: string
+
+  @Column({ type: 'json' })
+  modulesCompleted: string[]
+
+  @Column({ type: 'json' })
+  modulesDoing: string[]
+
+  @Column()
+  matriculationYear: number
+
+  @Column()
+  graduationYear: number
+
+  @Column()
+  graduationSemester: number
+
+  /**
+   * Given a module code, checks if user has cleared sufficient pre-requisites.
+   * Currently does not check for preclusion.
+   *
+   * @param{string} moduleCode
+   * @return{Promise<boolean>}
+   */
+  async canTakeModule(moduleCode: string): Promise<boolean | void> {
```

Comment:
This remains as part of the Entity because it is truly an instance-specific method